lib/world.js

```diff
@@ -1,22 +1,70 @@
 const Tray        = require('../lib/tray');
+const Gallery = require('./gallery');
+const Overlord = require('./overlord');
+const TrayCollision = require('./tray-collision');
+const ScoreBoard = require('./score-board');
+const JunkFood = require('./junk-food');
+const RandomJunkFood = require('./random-junk-food');
+const UserInput = require('./user-input');
 
 function World(width, height, ctx){
   this.width    = width;
   this.height   = height;
   this.ctx      = ctx;
   this.tray     = new Tray({ctx: ctx});
+  this.junkFoods = [];
 }
 
-World.prototype.rightArrow = function() {
-  if (this.tray.x <= 500) {
-    this.tray.moveRight();
-  }
+var gallery;
+var overlord;
+var overlordImage;
+var trayCollision;
+var startButton = document.getElementById('start');
+var scoreBoard = '';
+var gameRunning = false;
+var healthFoods = [];
+var overlordImage;
+var userInput;
+
+World.prototype.makeNewGame = function(){
```

Comment:
Rather than have the World make a new game, could you wrap your World object inside of a new Game object? The game object could continuously check whether the World has met requirements for the end of that instance, then report scores, return to the menu, start a new World, etc. The structure might look like:

Game.update()
if world is completed,  call end-game-sequence()
else
Game.world.update()

then you would have method(s)
end-game-sequence = function() {
end that world
fetch total score
report score
store in localStorage
return to Main Menu
}

I think this would be able to remove the prototype functions:
makeNewGame
setGameStyling
endGame
resetGame
endGameStyling
checkGameEnders